### PR TITLE
Implement gulp build process

### PR DESCRIPTION
**Pull request by alexmckenley**
@malucky @deini

There is still a bunch of work to be done, but this implements a gulp build process while maintaining the same functionality. 

I would recommend checkout this out before we merge.. just to be sure I didnt miss anything

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -4,6 +4,68 @@ var git = require('gulp-git');
...
+ * Compile templates for use in the templateCache.
+ */
+gulp.task('html', function () {
+    gulp.src(paths.html)
+        .pipe(templateCache('templates.js', {standalone: true}))
+        .pipe(gulp.dest('build/js/'));
+});
+
+/**
+ * Compile templates for use in the templateCache.
```
Comment (malucky) [inline gulpfile.js]: update comment

File: .gitignore [hunk 2]
```diff
@@ -39,4 +39,6 @@ node_modules
 ### WebStorm ###
 .idea/
 
+build
```
Comment (deini) [inline .gitignore:42]: Shouldn't this be `build/` ? or it's the same?

File: gulpfile.js [hunk 3]
```diff
@@ -4,6 +4,68 @@ var git = require('gulp-git');
...
+});
+
+/**
+ * Concat scripts and move to build dir.
+ */
+gulp.task('scripts', function() {
+  gulp.src(paths.scripts)
+    //.pipe(sourcemaps.init())
+      .pipe(concat('bundle.js'))
+    //.pipe(sourcemaps.write())
```
Comment (malucky) [inline gulpfile.js:48]: add a comment for sourcemap and fix spacing

File: .gitignore (hunk 2 above)
Comment (alexmckenley) [inline .gitignore:42]: I just used the same syntax as the `node_modules` folder on line 32. It should be the same AFAIK

Comment (deini) [inline .gitignore:42]: :+1:

File: public/index.html [hunk 4]
```diff
@@ -160,24 +156,9 @@ <h3 ng-show="buffering"><i class="fa fa-spinner fa-spin"></i></h3>
...
-  <script src="/scripts/services/scraper.js"></script>
-  <script src="/scripts/directives/myDirective.js"></script>
-  <script src="/scripts/controllers/frame.js"></script>
-  <script src="/scripts/controllers/queue.js"></script>
-  <script src="/scripts/controllers/search.js"></script>
-  <script src="/scripts/controllers/player.js"></script>
-  <script src="/scripts/controllers/share.js"></script>
-  <script src="/scripts/controllers/playlist.js"></script>
+  <script src="/js/templates.js"></script>
+  <script src="/js/bundle.js"></script>
```
Comment (deini) [inline public/index.html:160]: Beautiful :gem:

File: public/index.html [hunk 5]
```diff
@@ -160,24 +156,9 @@ <h3 ng-show="buffering"><i class="fa fa-spinner fa-spin"></i></h3>
...
+  <!-- <script src="/js/bundle.js"></script> -->
```
Comment (alexmckenley) [inline public/index.html]: I'll remove this

**Comment by alexmckenley** [comment]
Theres still a lot of work that needs to be done to make this cleaner, but at least this should allow us to have some sort of build strategy.

**Comment by alexmckenley** [comment]
I'm going to PR to the dev branch instead of master

**Pull request closed without merging**